Toolset/libraries/revinitialisationlibrary.livecodescript

```diff
@@ -16,18 +16,26 @@ command revInternal__LoadLibrary pLibrary
    end if
    
    // AL-2015-02-20: [[ IDE Restructure ]] Look for libraries in the correct locations.
-   set the itemDelimiter to "."
    if the environment begins with "development" then
       local tLibraryPath, tLibSuffix
+      set the itemDelimiter to "."
+      put the last item of pLibrary into tLibSuffix
       
       if not revEnvironmentIsInstalled() then
-         -- look for libraries in parent repository
-         if there is a file (revEnvironmentRepositoryPath() & "/ide-support/" & pLibrary & ".livecodescript") then
-            put revEnvironmentRepositoryPath() & "/ide-support/" & pLibrary & ".livecodescript" into tLibraryPath
+         -- look for libraries in parent development locations
+         local tRepoPath
+         put revEnvironmentRepositoryPath() into tRepoPath
+         if there is a file (tRepoPath & "/ide-support/" & pLibrary & ".livecodescript") then
+            put tRepoPath & "/ide-support/" & pLibrary & ".livecodescript" into tLibraryPath
          else 
-            put the last item of pLibrary into tLibSuffix
-            if there is a file (revEnvironmentRepositoryPath() & "/extensions/script-libraries/" & tLibSuffix & slash & tLibSuffix & ".livecodescript") then
-               put revEnvironmentRepositoryPath() & "/extensions/script-libraries/" & tLibSuffix & slash & tLibSuffix & ".livecodescript" into tLibraryPath
+            if there is a file (tRepoPath & "/extensions/script-libraries/" & tLibSuffix & slash & tLibSuffix & ".livecodescript") then
+               put tRepoPath & "/extensions/script-libraries/" & tLibSuffix & slash & tLibSuffix & ".livecodescript" into tLibraryPath
+            else
+               set the itemDelimiter to slash
+               delete the last item of tRepoPath
+               if there is a file (tRepoPath & "/extensions/script-libraries/" & tLibSuffix & slash & tLibSuffix & ".livecodescript") then
+                  put tRepoPath & "/extensions/script-libraries/" & tLibSuffix & slash & tLibSuffix & ".livecodescript" into tLibraryPath
+               end if
```

Comment:
@montegoulding I wonder if it would be worthwhile to tidy this code up a bit to improve readability and reduce duplication. What about something like:

```
if not revEnvironmentIsInstalled() then
   local tPath, tRepoPath, tFound
   put revEnvironmentRepositoryPath() into tRepoPath
   put false into tFound
   set the itemDelimiter to slash

   if not tFound then
      put merge("[[tRepoPath]]/ide-support/[[pLibrary]].livecodescript") into tPath
      put there is a file tPath into tFound
   end if
   if not tFound then
      put merge("[[tRepoPath]]/extensions/script-libraries/[[tLibSuffix]]/[[tLibSuffix]].livecodescript") into tPath
      put there is a file tPath into tFound
   end if
   if not tFound then
      delete the last item of tRepoPath
      put merge("[[tRepoPath]]/extensions/script-libraries/[[tLibSuffix]]/[[tLibSuffix]].livecodescript") into tPath
      put there is a file tPath into tFound
   end if
```

This still has some annoyingly unnecessary duplication though... 😞